test-app/tests/unit/models/sprite-snapshot-node-builder-test.ts

```diff
@@ -0,0 +1,436 @@
+import { module, test } from 'qunit';
+import SpriteTree from 'animations-experiment/models/sprite-tree';
+import AnimationContextComponent from 'animations-experiment/components/animation-context';
+import {
+  SpriteSnapshotNode,
+  SpriteSnapshotNodeBuilder,
+} from 'animations-experiment/models/sprite-snapshot-node-builder';
+import SpriteModifier from 'animations-experiment/addon/modifiers/sprite';
+
+class MockAnimationContext
+  implements
+    Pick<
+      AnimationContextComponent,
+      'element' | 'isStable' | 'isInitialRenderCompleted'
+    >
+{
+  id: string | undefined;
+  element: HTMLElement;
+  isAnimationContext = true;
+  isStable = true;
+  isInitialRenderCompleted = false;
+  currentBounds = new DOMRect(0, 0, 0, 0);
+  lastBounds = new DOMRect(0, 0, 0, 0);
+
+  constructor(
+    parentEl: HTMLElement | null = null,
+    id: string | undefined = undefined,
+    element: HTMLElement | null = null
+  ) {
+    this.element = element ?? document.createElement('div');
+    if (parentEl) {
+      parentEl.appendChild(this.element);
+    }
+    this.id = id;
+  }
+
+  captureSnapshot() {
+    console.log(
+      'no-op capture snapshot because we want to manually control bounds'
+    );
+  }
+
+  stable() {
+    this.isStable = true;
+    return this;
+  }
+
+  movedDownBy4() {
+    this.currentBounds = new DOMRect(0, 4, 0, 0);
+    return this;
+  }
+
+  unstable() {
+    this.isStable = false;
+    return this;
+  }
+}
+
+class MockSpriteModifier implements Partial<SpriteModifier> {
+  element: HTMLElement;
+  id: string;
+  currentBounds = new DOMRect(0, 0, 0, 0);
+  lastBounds = new DOMRect(0, 0, 0, 0);
+  isSpriteModifier = true;
```

Comment:
This is a hack for getting around
```
if (contextDescendant instanceof SpriteModifier) {
```
in SpriteSnapshotNodeBuilder's constructor.  Should find a way to do this without using `instanceof` before this can be merged.